tools/Voron.Recovery/Recovery.cs

```diff
@@ -259,22 +339,164 @@ public RecoveryStatus Execute(CancellationToken ct)
                         mem = PrintErrorAndAdvanceMem(message, mem, logFile);
                     }
                 }
+                attachmentWriter.WriteEndArray();
                 documentsWriter.WriteEndArray();
                 conflictsWriter.WriteEndArray();
                 revisionsWriter.WriteEndArray();
+                attachmentWriter.WriteEndObject();
                 documentsWriter.WriteEndObject();
                 conflictsWriter.WriteEndObject();
                 revisionsWriter.WriteEndObject();
 
+                ReportOrphanAttachmentsAndMissingAttachments(logFile, ct);
                 logFile.WriteLine(
                     $"Discovered a total of {_numberOfDocumentsRetrieved:#,#;00} documents within {sw.Elapsed.TotalSeconds::#,#.#;;00} seconds.");
+                logFile.WriteLine($"Discovered a total of {_attachmentsHashs.Count:#,#;00} attachments.");
                 logFile.WriteLine($"Discovered a total of {_numberOfFaultedPages::#,#;00} faulted pages.");
             }
             if (_cancellationRequested)
                 return RecoveryStatus.CancellationRequested;
             return RecoveryStatus.Success;
         }
 
+        private void ReportOrphanAttachmentsAndMissingAttachments(StreamWriter logFile, CancellationToken ct)
+        {
+            //No need to scare the user if there are no attachments in the dump
+            if (_attachmentsHashs.Count == 0 && _documentsAttachments.Count == 0)
+                return;
+            if (_attachmentsHashs.Count == 0)
+            {
+                logFile.WriteLine("No attachments were recoverd but there are documents pointing to attachments.");
```

Comment:
Add count of attachments and a list of documents missing attachments.

Reply:
This can be an _old_ document that was deleted, and its attachments as well, and then got reused.
So that is a valid scenario for us, but we need to give the user this information, and explain why this can happen.